qt/scientific_interfaces/Indirect/IndirectFitPropertyBrowser.cpp

```diff
@@ -0,0 +1,394 @@
+  connect(m_functionBrowser,
+          SIGNAL(parameterChanged(const QString &, const QString &)), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser, SIGNAL(tiesChanged()), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser, SIGNAL(constraintsChanged()), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser, SIGNAL(globalsChanged()), this,
+          SIGNAL(functionChanged()));
+  connect(m_functionBrowser,
+          SIGNAL(localParameterButtonClicked(const QString &)), this,
+          SIGNAL(localParameterEditRequested(const QString &)));
+}
+
+void IndirectFitPropertyBrowser::initFitOptionsBrowser() {
+  m_fitOptionsBrowser = new FitOptionsBrowser(
+      nullptr, FitOptionsBrowser::SimultaneousAndSequential);
+  m_fitOptionsBrowser->setObjectName("fitOptionsBrowser");
+  m_fitOptionsBrowser->setCurrentFittingType(FitOptionsBrowser::Sequential);
+}
+
+bool IndirectFitPropertyBrowser::isFullFunctionBrowserActive() const {
+  return m_functionWidget->currentIndex() == 1;
+}
+
+MultiDomainFunction_sptr IndirectFitPropertyBrowser::getGlobalFunction() const {
+  auto fun = isFullFunctionBrowserActive()
+                 ? m_functionBrowser->getGlobalFunction()
+                 : m_templateBrowser->getGlobalFunction();
+  return boost::dynamic_pointer_cast<MultiDomainFunction>(fun);
+}
+
+IFunction_sptr IndirectFitPropertyBrowser::getSingleFunction() const {
+  return isFullFunctionBrowserActive() ? m_functionBrowser->getFunction()
+                                       : m_templateBrowser->getFunction();
+}
+
+QStringList IndirectFitPropertyBrowser::getGlobalParameters() const {
+  return isFullFunctionBrowserActive()
+             ? m_functionBrowser->getGlobalParameters()
+             : m_templateBrowser->getGlobalParameters();
+}
+
+QStringList IndirectFitPropertyBrowser::getLocalParameters() const {
+  return isFullFunctionBrowserActive()
+             ? m_functionBrowser->getLocalParameters()
+             : m_templateBrowser->getLocalParameters();
+}
+
+void IndirectFitPropertyBrowser::syncFullBrowserWithTemplate() {
+  auto const fun = m_templateBrowser->getFunction();
+  m_functionBrowser->setFunction(fun);
+  if (fun) {
+    m_functionBrowser->updateMultiDatasetParameters(
+        *m_templateBrowser->getGlobalFunction());
+    m_functionBrowser->setGlobalParameters(
+        m_templateBrowser->getGlobalParameters());
+  }
+}
+
+void IndirectFitPropertyBrowser::syncTemplateBrowserWithFull() {
+  auto const funStr = m_functionBrowser->getFunctionString();
+  m_templateBrowser->setFunction(funStr);
+  if (auto const fun = m_functionBrowser->getGlobalFunction()) {
+    m_templateBrowser->updateMultiDatasetParameters(*fun);
+    m_templateBrowser->setGlobalParameters(
+        m_functionBrowser->getGlobalParameters());
+  }
+}
+
+void IndirectFitPropertyBrowser::init() {
+  initFunctionBrowser();
+  initFitOptionsBrowser();
+
+  auto w = new QWidget(this);
```

Comment:
Ditto on new

Reply:
This is a child of the main widget so lifetime is handled by that